group-aws/utility-modules.md

```diff
@@ -0,0 +1,104 @@
+
+For this reason the utility modules are expected to have a smaller amount of higher quality code. 
+
+Criteria for putting functions into the module_utils/aws directory 
+
+* Used in at least two modules
+* Reasonably clear, sensible and agreed upon interface
+* Don't know of a reason to think it will become obsolete soon
+* 90% or more unit test coverage (line)
+
+## Planned Layout
+
+The layout is subject to change, probably until just before the 2.5 release (at which point there
+should be a stable interface document).  Modules included in Ansible will get refactored together
+with the change.  Pull requests will have to be refactored before being accepted.  Modules ourside
+the main repository which are already using this interface will have to be refactored by their
+authors.
+
+    aws/
+        core.py - Basic AWS modules 
+            class AnsibleAWSModule - class to use for handling module set up'; based off AnsibleModule
+	    	.params() - returns the parameter of the module
+		.exit_json() - low level module exit function from AnsibleModule
+		.fail_json() - low level module exit with failure function from AnsibleModule
+		.fail_json_aws(exception, msg=<string>) - higer level function for exiting after exceptions
+		.get_aws_connection_info() - return information about AWS connection;  you don't need to call this just to connect
+                .get_aws_session() - returns a low level botocore session
+                .get_aws_client() - returns a boto3 medium level client *** check this
+                .get_aws_resource() - returns a high level boto3 resource
+            class AWSRetry (not yet moved in)
+            def camel_dict_to_snake_dict(camel_dict)
+        direct_connect.py
+        iam.py 
+            def convert_friendly_names_to_arns
+        rds.py (names still subject to change - please comment in [PR](https://github.com/ansible/ansible/pull/26594)) 
+		get_rds_instance() -> rds_camel_dictionary
+		rds_instance_to_facts(camel_dict) -> rds_facts_snake_dict()
+		get_rds_snapshot() -> snapshot_camel_dictionary
+		rds_snapshot_to_facts(camel_dict) -> rds_facts_snake_dict()
+        resource.py - functions related to any AWS resources
+            def wait_for_status(resource, status)
+        vpc.py
+
+Please note that various functions not listed above which are included in AnsibleModule may not be
+replicated by AnsibleAWSModule.  This is deliberate to allow future simplification of the base
+class of AnsibleAWSModule.  If you have a good reason for making other functions available please
+discuss this with the AWS working group who should be happy to help or at least explain why the
+function isn't available.
+
+## Using the module utilities
+
+*This section is temporary until the module utilities become standard and should be moved to the*
+*[AWS module guildelines in the amazon modulesdirectory](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md)*
```

Comment:
typo: modules directory.

Worth splitting this line too - I know the url will make at least one line longer but still makes review easier